### DISCUSSION TITLE
Video about Meshtastic LoRa messaging device

### DISCUSSION THREAD

**Discussion by trosel** [markqvist/Reticulum#487] [General]
https://www.youtube.com/watch?v=40llxjrIG3w

Isn't this little device the same thing as the RNode? https://unsigned.io/rnode/

Biggest difference is that a lot more love has been given to the client apps on Meshtastic. Something [I would love to see happen here](https://github.com/markqvist/Reticulum/discussions/388).

**Comment by faragher** [comment]
They're on similar hardware and use LoRa, but really the similarities end there.

To my understanding, Meshtastic is a stand-alone unit you connect to and send messages.

The RNode is a modem. It requires the actual work to be done on the computer/phone.

For a lot of people, Mesthastic does everything they need: it sends local messages that get repeated and can be seen by everyone in the group. It's a great tool for that scope and use case.

The RNode connects a Reticulum network over LoRa. This is an important distinction because it's a very different use case and serves different needs.  Reticulum runs on a computer, while the RNode simply provides an interface (the modem can also be run directly by serial applications).

Regarding client apps, I'd love to see more, but nobody's stepped up to the plate to make any. That's the issue with open-source systems: they grow based on the developers involved. Since the network stack is currently considered feature complete, it's easy for a skilled developer to make an LXMF app in Python, but you also need someone who knows how to create the exact kind of app you're looking for. If you know anyone with those skills and the desire to help, send them our way. 

Given the increasing featureset of Sideband, it might be worth revisiting to see if it works as a mobile app. That being said, at this time the RNS doesn't run on embedded systems, so an all-in-one turnkey piece of hardware isn't possible with ESP32 boards. You could get things running with a Pi Zero or similar, but the requirements are vastly different.

**Reply by trosel** [reply]
>The RNode is a modem. It requires the actual work to be done on the computer/phone.

According to this, I don't think that's true: https://github.com/markqvist/Reticulum/discussions/388#discussioncomment-7449060

**Reply by faragher** [reply]
That is absolutely the case. You can check the firmware yourself to confirm: the network stack doesn't run on the RNode, it's simply a modem.

**Reply by trosel** [reply]
Is @markqvist wrong then?

> Just to make it clear, you can totally make 2 RNodes (or buy them from someone), install Sideband on you and your friends phone, and start messaging each other directly

**Reply by faragher** [reply]
The phone application runs the network stack. The RNode still only acts as a modem.

**Reply by trosel** [reply]
I think my confusion then was with your comment

> This is an important distinction because it's a very different use case and serves different needs. 

It's the same use case and same need in this example scenario. Reticulum just has a thick client while meshtastic has a thin client.

**Reply by faragher** [reply]
You still appear to be confused. It's like saying a smart TV has the same use case as a monitor attached to a desktop computer. 

Both Meshtastic and Reticulum can send messages so a local source. The RNode is a modem that Reticulum can use. That was your initial question, if the hardware looks the same, doesn't it do the same thing? 

Now, Meshtastic can send messages to groups across a few hops without routing. It's basically an Ethernet hub. As with a home network, it's great for small systems where you can't be sure whats where.

Reticulum is like an Ethernet router: it actively directs where a packet goes. It can also support arbitrary data lengths, run over LoRa, Ethernet, serial, printed documents, RF, and basically any medium available.

If someone wants to communicate with a local group over a mesh network, Meshtastic works great, and is easier to get into, especially for group messaging.

If you want to chat with individuals over end to end encrypted channels, Sideband or NomadNet are great choices. If you want to make applications or send arbitrary data, using Reticulum directly is a far better choice.

For the use case of talking with someone over LoRa, both are useful. For group messaging over LoRa in a small area (like a city) then Meshtastic is a better choice. For point to point secure messaging over any medium (or across multiple media) over any distance, delayed or Sneakernet messaging, or machine to machine traffic, Reticulum is the better option. 

Saying they have the same use case will only cause you to be disappointed.

**Reply by trosel** [reply]
> Saying they have the same use case will only cause you to be disappointed.

As a user, I want to talk to friends over LoRa. They can both do that. Reticulum is a thick client while Meshtastic is a thin client. 

Am I really confused?

**Reply by markqvist** [reply]
> Am I really confused?

Yes. The thick/thin client analogy makes no sense here. Re-read what faragher already posted, I couldn't really add anything.

If you are still confused about it all, I'd recommend reading up on what a networking stack is, and some of the fundamentals of how digital communication works.

**Reply by trosel** [reply]
@markqvist is this true or not? https://github.com/markqvist/Reticulum/discussions/388#discussioncomment-7449060

If it is true, then you can do the same thing with RNode + Sideband that you can in the Meshtastic video.

**Reply by markqvist** [reply]
Yes, the Sideband app paired with an RNode has been able to do the same thing as meshtastic for a long time now.

It can also do a lot more, like:

- File transfers
- Sending and receiving images
- Infrastructureless location updates and telemetry
- Remote commands and requests
- Has a fully fledged plugin system
- Can run on both Android, Linux, macOS and Windows
- Can use *any* medium to communicate, not just LoRa
- Has unlimited network size
- Has unlimited amount of endpoints and messaging adresses
- Has global reachability
- Has complete and universal self-ownership of addresses
- Does not depend on any internet-based servers to function
- Has offline, delay-tolerant and failure-resilient distributed message storage

**Reply by trosel** [reply]
> It can also do a lot more

That's great! So what's the next step? Just getting users, right?
